boxsdk/object/base_object.py

```diff
@@ -46,22 +30,14 @@ def __init__(self, session, object_id, response_object=None):
         :type response_object:
             :class:`BoxResponse`
         """
-        super(BaseObject, self).__init__(session)
         self._object_id = object_id
-        self._response_object = response_object or {}
-        self.__dict__.update(self._response_object)
-
-    def __getitem__(self, item):
-        """Base class override. Try to get the attribute from the API response object."""
-        return self._response_object[item]
+        # super(BaseObject, self).__init__(session, response_object)
+        BaseEndpoint.__init__(self, session, response_object)
+        BaseAPIJSONObject.__init__(self, response_object)
 
-    def __repr__(self):
-        """Base class override. Return a human-readable representation using the Box ID or name of the object."""
-        description = '<Box {0} - {1}>'.format(self.__class__.__name__, self._description)
-        if six.PY2:
-            return description.encode('utf-8')
-        else:
-            return description
+    def __eq__(self, other):
```

Comment:
If you move this back to the position it used to be in, then it won't appear in the diff, simplifying things. Same with `get_url` and `get_type_url`.